### PR TITLE
DRAFT Fix StridedSlice with -1 shape

### PR DESCRIPTION
**Pull request by seanshpark**
on-going draft to fix StridedSlice with -1 shape.

Signed-off-by: SaeHie Park <saehie.park@gmail.com>

### CODE REVIEW

File: res/TensorFlowLiteRecipes/StridedSlice_004/test.recipe [hunk 1]
```diff
@@ -0,0 +1,50 @@
+#
+# Failed case from https://github.com/Samsung/ONE/issues/9439
+#
+operand {
+  name: "Placeholder"
+  type: FLOAT32
+  shape { dim: 1 dim: 16 dim: 16 dim: 32 }
+  quant { quantized_dimension: 0 }
```
Comment (jinevening) [inline res/TensorFlowLiteRecipes/StridedSlice_004/test.recipe:8]: The target model is fp32 model, so `quant` field is not necessary for all operands.

Comment (seanshpark) [reply res/TensorFlowLiteRecipes/StridedSlice_004/test.recipe:8]: `for all operands.`

Can you confirm for `INT32` too?
- add `quant` for UINT8 and INT16 only
- if and only if `quantized_dimension` is not zero

And what about `scale` and `zero_point` ?

Comment (jinevening) [reply res/TensorFlowLiteRecipes/StridedSlice_004/test.recipe:8]: > can you confirm for INT32 too?
> And what about scale and zero_point ?

For the target model, quantization parameters (`quant`) is not necessary for INT32 tensors. scale and zero_point are not necessary as well.

As to https://github.com/Samsung/ONE/issues/9446 (not directly related to the strided slice model in this PR), 

I think tflchef-reverse should generate quantization parameters (`quant`) based on the tflite model, i.e., `quantization` field in the `Tensor` (from tflite schema)

```
  quantization:QuantizationParameters;  // Optional.
```

We may not need complex rules for generating quantization parameters.

**Comment by seanshpark** [comment]
@jinevening , can you please check with current draft?

**Comment by jinevening** [comment]
I checked that `circle2circle` works well for the target model on this DRAFT. 👍

**Comment by seanshpark** [comment]
done

**Pull request closed without merging**